doc/develop/test/twister.rst

```diff
@@ -408,6 +408,16 @@ platform_allow: <list of platforms>
     constraints, this option should only be used if the test or sample can
     only be run on the allowed platform and nothing else.
 
+    .. warning::
+
+      Some platform enables the testing for themselves by default, the test scenario **will** be run
+      for those platforms if you don't explicitly specify an allowed platform.
+
+      Also, if you specify one of those platforms that enable testing by default and another one
+      that doesn't, the latter will not be run unless you ask Twister to select it when running
+      Twister with ``--platform``. That is because Twister will select the platform that are both in
+      the default list of allowed platform and the list of allowed platform you specified.
```

Comment:
Note that:
1. the `allowed_platforms` list acts as an AND-style filter **only if** it contains at least one of the boards that has enabled testing by default. In that case, twister will *not* run the test on other platforms in that list that have not enabled testing by default. 
2. When none of the boards in the `allowed_platforms` list have enabled testing by default, then twister will run the test on all of them.

In 1., to force running on all platforms specified in the `allowed_platforms` list, use the ``--platform`` option.